### PR TITLE
tests.acceptance: login once per session

### PR DESCRIPTION
**Pull request by david-caro**
Addresses #1987

### CODE REVIEW

**Comment by david-caro** [comment]
It seems to cut out ~40 secs...

**Comment by jacquerie** [comment]
> It seems to cut out ~40 secs...

Sounds promising, but the duration of this test suite has variance which is much higher than that, so one sample does not really prove it...

**Comment by david-caro** [comment]
That's where the "seems" comes from ;)


--
David Caro <david.caro@cern.ch>

CERN - RCS-SIS
inSPIRE-HEP High Energy Physics information system
http://inspirehep.net
On May 11, 2017 22:31, Jacopo Notarstefano <notifications@github.com> wrote:

It seems to cut out ~40 secs...

Sounds promising, but the duration of this test suite has variance which is much higher than that, so one sample does not really prove it...

File: tests/acceptance/conftest.py [hunk 1]
```diff
@@ -74,14 +76,33 @@ def app(request):
...
+    if os.path.exists(cookies_path):
+        os.unlink(cookies_path)
 
 
 @pytest.fixture
-def login(arsenic):
-    top_navigation_page.log_in('admin@inspirehep.net', '123456')
-    yield
-    top_navigation_page.log_out()
+def login(arsenic, app):
```
Comment (jacquerie) [inline tests/acceptance/conftest.py]: The `arsenic` fixture already depends on `app`, so this shouldn't be needed.

File: tests/acceptance/conftest.py [hunk 2]
```diff
@@ -74,14 +76,33 @@ def app(request):
...
+    cookies_path = './tests/acceptance/login_cookies.pkl'
+    if not os.path.exists(cookies_path):
+        top_navigation_page.log_in('admin@inspirehep.net', '123456')
+        sleep(1)
```
Comment (jacquerie) [inline tests/acceptance/conftest.py]: Is this needed, or is it just to be safe?

File: tests/acceptance/conftest.py [hunk 3]
```diff
@@ -74,14 +76,33 @@ def app(request):
 
 @pytest.fixture
 def arsenic(selenium, app):
-    Arsenic(selenium)
+    return Arsenic(selenium)
```
Comment (jacquerie) [inline tests/acceptance/conftest.py]: Oh, so this is why we have `Arsenic()` everywhere?

File: .gitignore [hunk 4]
```diff
@@ -67,6 +67,7 @@ nosetests.xml
 coverage.xml
 *,cover
 .hypothesis/
+tests/acceptance/login_cookies.pkl
```
Comment (jacquerie) [inline .gitignore]: This is not really a coverage report, so maybe we should add an extra section for `# Application specific ignores` or something like that.

**Pull request closed without merging**